### ISSUE TITLE
Images are labelled in compressed clusters when using own cfg and weights

### ISSUE THREAD

**Issue by raggot** [ayooshkathuria/pytorch-yolo-v3#15]
Hi,

After I successfully loaded my own `yolov3-voc-own.weights` obtained from training the network with Pascal VOC, referred to the `yolov3-voc.cfg` I used for training file, and modified classes names and number (from `coco.names` to `voc.names` and from `80` to `20`) I was surprised at the results.

Please have a look at the following images
Herd of horses: https://ibb.co/iO8WFo
Messi/Inista: https://ibb.co/nMNaao
Guy with poncho and dog: https://ibb.co/j5bU1T

In all of them, there is a pattern of identifying the right combination of images (incorrectly, but clearly finding _something_) but in the wrong location, and condensed. My guess is that the recognition happens in one of the 'zoomed' version of the image, and it's identified anchors are not correctly remapped to the original size of the image.

Is this any reasonable? Do you think this could be a bug from the 'interpretation' of the `.cfg` file? Or perhaps I'm missing another piece of the puzzle?

Another possible point of code that could possibly be related to this is the input argument of scales, where I feed `1,2,3` as default. Could it be that it should be matched with some properties of my network?

Thanks in advance!

**Comment by ayooshkathuria** [comment]
Can you upload your cfg and weights file somewhere so that I can have a look? Google drive maybe?

**Comment by ayooshkathuria** [comment]
And can you confirm things are working fine with the yolo-v3.cfg and original weightsfile?

**Comment by raggot** [comment]
It's impressive how fast you respond. I confirm everything work in the script using your default configuration.

As per the files, I did as you suggested:
This is the [cfg](https://drive.google.com/file/d/1sXc9Ui2AVETj3IOKwkoBnh3840sIcGVW/view?usp=sharing)
And this is the [weight file](https://drive.google.com/file/d/1giY7kBuDlkh2oogGuslVnDZmGNjoAccY/view?usp=sharing)

Thank you.

**Comment by ayooshkathuria** [comment]
Something is off about the weights file. Everything is working fine. The raw "un-transformed" results that are being produced are not correct. I would have dug deeper but I'm currently looking for a job/research fellowship and will be really busy with that. If anything, this implementation was done so as to drive my research, and I'd be focusing exclusively on that, and getting the training code ready so as to further my research experiments. So, I won't be able to give too much time to the issue. I'll try to train the data on PASCAL VOC once I have the training code, but till then I won't be able to do much. You can check out the blog posts on Paperspace I've linked in the repo readme where I've explained in detail the implementation is. Or you can just ask me here what each part of the code does and I'll try my best to explain stuff.

**Comment by raggot** [comment]
Do you just mind telling more precisely what you mean with this:

> Everything is working fine. The raw "un-transformed" results that are being produced are not correct. 

Which part of the code do you think is the problem? If it's on the image handling side I can maybe fix it myself and create a pull request. If not, if you still have an indication of where it could be (even at a lower level) I can try digging anyway although I'm not very familiar with NNs. 

A reference to your Paperspace would also be nice, so I can understand better where to put my hands.

Thanks.

**Comment by ayooshkathuria** [comment]
I'm sorry for not replying for so long. Have been super busy. Will try to look into this issue on weekends. By raw "un-transformed" results mean't I wrote manual code to transform the output feature map to bounding box on images, and no fault there. It seems as if the problem is with weights, because the problem seems to lie with output of the network and not the pipeline that turns the output into detections. 

You can read in-depth about my implementation here. 

https://blog.paperspace.com/how-to-implement-a-yolo-object-detector-in-pytorch/

**Comment by raggot** [comment]
@ayooshkathuria, I believe you are right. I later realised my training process was garbage. I was unable to train with the default cfg of yolo3 because of memory limitations on my GPU, and inexperienced manual modifications blewed it. I found curious that my results still clustered realistic detections given the image, but perhaps I was fooled by randomness.

I ended up abandoning yolo3 and switch to yolo2. Along with that, I also switched to another open-source project. 

Keep being busy with more important matters of life. Thanks for your time and sharing of your code.
